app/models/non_primitive_property.rb

```diff
@@ -8,4 +8,12 @@ class NonPrimitiveProperty < ApplicationRecord
   has_one_attached :attachment
 
   validates_presence_of :label
+
+  before_save :reset_allow_attachments
+
+  private
+
+  def reset_allow_attachments
```

Comment:
@alis-khadka what does this function do? This is something I [struggled to understand in the original revision of this PR as well](https://github.com/restarone/violet_rails/pull/837/files#diff-b2d2563b673e233199615fe938bc7b5f3f79b975ce5b3dcb6df4be3ef4bb3c20)

Reply:
Since the toggling of attachment behavior is only meant for **richtext** type, it is resetting the boolean value for other field types.

Reply:
@alis-khadka I see, can't we set this to `default: false` in the migration + add a backfill and not have this callback?